zavod/zavod/dedupe.py

```diff
@@ -22,33 +21,26 @@
 AUTO_USER = "zavod/xref"
 
 
-@cache
-def get_resolver() -> Resolver[Entity]:
+def _get_resolver() -> Resolver[Entity]:
     """Load the deduplication resolver."""
     if settings.RESOLVER_PATH is None:
         raise RuntimeError("Please set $ZAVOD_RESOLVER_PATH.")
     log.info("Loading resolver from: %s" % settings.RESOLVER_PATH)
     return Resolver.load(Path(settings.RESOLVER_PATH))
 
 
-def get_dataset_resolver(dataset: Dataset) -> Resolver[Entity]:
-    """Get a resolver for the given dataset."""
-    if not dataset.resolve:
-        return Resolver()
-    return get_resolver()
+@cache
+def get_resolver() -> Resolver[Entity]:
+    """Load the deduplication resolver."""
+    return _get_resolver()
 
 
+@cache
 def get_dataset_linker(dataset: Dataset) -> Linker[Entity]:
```

Comment:
Would it make sense to hoist the `dataset.resolve` check out of this function? As far as I can tell we're loading the same resolver for all datasets, so there's no reason for breaking the cache for different datasets

Reply:
Maybe it's actually safer to get rid of the cache entirely

Reply:
Yeah, if we know to only create the linker once.